### PR TITLE
Problem: Vendored NFT module's permissions are too lax

### LINKED ISSUE #520: Problem: vendored NFT module's permissions are too lax
> By default it will not fail if the transfer is executed by someone beside the owner. It is highly recommended that a custom handler is made to restrict use of this Message type to prevent unintended use.

The code seems to check for the owner==sender, but it allows the owner to modify the token content:

this should be modified to match similar business logic to CW721 (only the token issuer can edit and burn its tokens; the token owner can only transfer): spec + implementation need to be updated + integration tests for this expected logic

### PR DESCRIPTION
**Pull request by devashishdxt**
Solution: Changed the implementation as per new specs. Fixes #520.

### CODE REVIEW

File: proto/nft/v1/tx.proto [hunk 1]
```diff
@@ -45,11 +45,8 @@ message MsgTransferNFT {
 
     string id = 1;
     string denom_id = 2 [ (gogoproto.moretags) = "yaml:\"denom_id\"" ];
-    string name = 3;
-    string uri = 4 [ (gogoproto.customname) = "URI" ];
-    string data = 5;
-    string sender = 6;
-    string recipient = 7;
+    string sender = 3;
+    string recipient = 4;
```
Comment (tomtau) [inline proto/nft/v1/tx.proto:49]: not a protobuf expert -- if this was meant to be still somewhat compatible with the original IRISnet module... what's the best way when the fields are removed?

Comment (tomtau) [reply proto/nft/v1/tx.proto:49]: (if possible, not a major blocker)

File: x/nft/keeper/denom.go [hunk 2]
```diff
@@ -64,3 +64,21 @@ func (k Keeper) GetDenoms(ctx sdk.Context) (denoms []types.Denom) {
...
+// nolint: interfacer
+func (k Keeper) IsDenomCreator(ctx sdk.Context, denomID string, address sdk.AccAddress) (types.Denom, error) {
+	denom, err := k.GetDenom(ctx, denomID)
+	if err != nil {
+		return types.Denom{}, err
+	}
+
+	addressStr := address.String()
+
+	if addressStr != denom.Creator {
```
Comment (tomtau) [inline x/nft/keeper/denom.go:79]: I'd probably add a comment here why the string comparison is ok -- is the denom.creator guaranteed/checked to be a valid address or is it possible one can set it to invalid string (empty, some random...)?

**Review by brianatcrypto — APPROVED** [review]
changes look sensible.

**Review by tomtau — APPROVED** [review]

File: proto/nft/v1/tx.proto (hunk 1 above)
Comment (devashishdxt) [reply proto/nft/v1/tx.proto:49]: not changing the index?

File: x/nft/keeper/denom.go (hunk 2 above)
Comment (devashishdxt) [reply x/nft/keeper/denom.go]: Changed to comparing concrete address types

File: proto/nft/v1/tx.proto (hunk 1 above)
Comment (yihuang) [reply proto/nft/v1/tx.proto:49]: with protobuf rules, if you want to keep compatibility, never remove fields

Comment (tomtau) [reply proto/nft/v1/tx.proto:49]: maybe no need -- the end goal will be to migrate to standard Cosmos SDK module... for compat with IRISnet -- the response data didn't change, so browsers/wallets should be able to query / view the same things?

**Review by yihuang — APPROVED** [review]

**Review by tomtau — APPROVED** [review]

**Review by matias904 — APPROVED** [review]

**Pull request merged**